python/ray/data/_internal/execution/operators/map_transformer.py

```diff
@@ -45,10 +45,11 @@ def __init__(
         self._input_type = input_type
         self._output_type = output_type
 
+    @abstractmethod
     def __call__(
         self, input: Iterable[MapTransformFnData], ctx: TaskContext
     ) -> Iterable[MapTransformFnData]:
-        return self._callable(input, ctx)
+        pass
```

Comment:
can we do `raise NotImplementedError` instead?

Reply:
When using `@abstractmethod`, no need to add `raise NotImplementedError`. 
Abstract classes are better than NotImplementedError, because an exception will raised when creating the object, instead of when calling the method.

Reply:
got it, thanks!

Reply:
Nit: ... seems to be recommended practice https://docs.python.org/3/library/abc.html